testing/acceptance_test.py

```diff
@@ -645,6 +647,69 @@ def join_pythonpath(*dirs):
             "*1 passed*"
         ])
 
+    @pytest.mark.skipif(not hasattr(os, "symlink"), reason="requires symlinks")
+    def test_cmdline_python_package_symlink(self, testdir, monkeypatch):
+        """
+        test --pyargs option with packages with path containing symlink can
+        have conftest.py in their package (#2985)
+        """
+        monkeypatch.delenv('PYTHONDONTWRITEBYTECODE', raising=False)
+
+        search_path = ["lib", os.path.join("local", "lib")]
+
+        dirname = "lib"
+        d = testdir.mkdir(dirname)
+        foo = d.mkdir("foo")
+        foo.ensure("__init__.py")
+        lib = foo.mkdir('bar')
+        lib.ensure("__init__.py")
+        lib.join("test_bar.py"). \
+            write("def test_bar(): pass\n"
+                  "def test_other(a_fixture):pass")
+        lib.join("conftest.py"). \
+            write("import pytest\n"
+                  "@pytest.fixture\n"
+                  "def a_fixture():pass")
+
+        d_local = testdir.mkdir("local")
+        symlink_location = os.path.join(str(d_local), "lib")
+        if six.PY2:
+            os.symlink(str(d), symlink_location)
+        else:
+            os.symlink(str(d), symlink_location, target_is_directory=True)
+
+        # The structure of the test directory is now:
+        # .
+        # ├── local
+        # │   └── lib -> ../lib
+        # └── lib
+        #     └── foo
+        #         ├── __init__.py
+        #         └── bar
+        #             ├── __init__.py
+        #             ├── conftest.py
+        #             └── test_bar.py
+
+        def join_pythonpath(*dirs):
+            cur = os.getenv('PYTHONPATH')
+            if cur:
+                dirs += (cur,)
+            return os.pathsep.join(str(p) for p in dirs)
+
+        monkeypatch.setenv('PYTHONPATH', join_pythonpath(*search_path))
+        for p in search_path:
+            monkeypatch.syspath_prepend(p)
```

Comment:
@cryvate 
Is the different order between `PYTHONPATH` and `sys.path` intentional here?
Fixed it in https://github.com/pytest-dev/pytest/pull/4106/commits/2f058adaa6df7b5e0d27ced9c759570997150372, which is required to fix the test failure when using `realpath()` on the args (https://github.com/pytest-dev/pytest/pull/4108/commits/18e7358042d7dc16a9599f5ab57fdd88d5d1705e), but have not really investigated why.

Reply:
@blueyed 
Though git blame will assign the 'blame' to me, actually this part of the code was a copy-paste from the `test_cmdline_python_namespace_package` test so I'd pass on the 'blame' to this [PR](https://github.com/pytest-dev/pytest/commit/e2e6e317118d6d15ccdf6f6708a48c08d85cbc25). Regardless, I'd think that the different order (the opposite way around if I'm reading it correctly) is accidental here as well as in the other test.

Reply:
@cryvate 
Thanks and sorry for not looking good enough.

>  as well as in the other test

Which one do you mean?

Reply:
@blueyed 
Not a problem at all! I meant `test_cmdline_python_namespace_package` where I copy-pasted from.